drivers/gpio/gpio_imx.c

```diff
@@ -12,13 +12,21 @@
 #include <soc.h>
 #include <sys/util.h>
 #include <gpio_imx.h>
+#include <string.h>
+#ifdef CONFIG_PINCTRL
+#include <drivers/pinctrl.h>
+#endif
```

Comment:
I've been nacking any pinctrl driver making use of GPIO, but here I see the gpio driver making use of pinctrl for the first time. Can you please justify why the GPIO driver needs to make use of pinctrl infrastructure? While there's overlap  between GPIO/pinctrl (check e.g. GD32 or STM32), drivers should be kept independent unless we make a decision to share code between the two.

Reply:
I took this approach for the i.MX parts because their pin mux controller does not use a PORT,PIN type of architecture- so for example, GPIO1 pin 29 corresponds to some pad on the SOC, but that register may be anywhere in the IOMUXC peripheral register space. Moreover, each SOC has different numbers of GPIO pins present on each port.

The easiest way I could find to set these pinmux settings was to leverage pin control nodes to determine which register needs to be updated to select the GPIO mux for that SOC's pad.